src/java/org/apache/cassandra/service/CassandraDaemon.java

```diff
@@ -803,6 +809,21 @@ public static void registerNativeAccess() throws javax.management.NotCompliantMB
         MBeanWrapper.instance.registerMBean(new StandardMBean(new NativeAccess(), NativeAccessMBean.class), MBEAN_NAME, MBeanWrapper.OnException.LOG);
     }
 
+    @VisibleForTesting
+    public static void registerAsyncProfiler() throws javax.management.NotCompliantMBeanException
+    {
+        AsyncProfiler asyncProfiler;
+        if (!ASYNC_PROFILER_ADVANCED_MODE.getBoolean())
```

Comment:
@bbotella would not it be better if this was called `ASYNC_PROFILER_UNSAFE`? I do not know what "advanced" 
means. 

This seems to be way more straightforward.

`AsyncProfiler asyncProfiler = ASYNC_PROFILER_UNSAFE.getBoolean() ? new AsyncProfilerUnsafe() : new AsyncProfilerSafe();`